homeassistant/components/vera.py

```diff
@@ -0,0 +1,116 @@
+"""
+Support for Vera devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/vera/
+"""
+import logging
+
+from collections import defaultdict
+from requests.exceptions import RequestException
+
+from homeassistant import bootstrap
+from homeassistant.const import (
+    ATTR_SERVICE, ATTR_DISCOVERED,
+    EVENT_HOMEASSISTANT_STOP, EVENT_PLATFORM_DISCOVERED)
+from homeassistant.loader import get_component
+
+REQUIREMENTS = ['pyvera==0.2.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'vera'
+
+VERA_CONTROLLER = None
+
+CONF_EXCLUDE = 'exclude'
+CONF_LIGHTS = 'lights'
+
+BINARY_SENSOR = 'binary_sensor'
+SENSOR = 'sensor'
+LIGHT = 'light'
+SWITCH = 'switch'
+
+DEVICE_CATEGORIES = {
+    'Sensor': BINARY_SENSOR,
+    'Temperature Sensor': SENSOR,
+    'Light Sensor': SENSOR,
+    'Humidity Sensor': SENSOR,
+    'Dimmable Switch': LIGHT,
+    'Switch': SWITCH,
+    'Armable Sensor': SWITCH,
+    'On/Off Switch': SWITCH,
+    # 'Window Covering': NOT SUPPORTED YET
+}
+
+DISCOVER_BINARY_SENSORS = 'vera.binary_sensors'
+DISCOVER_SENSORS = 'vera.sensors'
+DISCOVER_LIGHTS = 'vera.lights'
+DISCOVER_SWITCHES = 'vera.switchs'
+
+VERA_DEVICES = defaultdict(list)
+
+
+# pylint: disable=unused-argument, too-many-function-args
+def setup(hass, base_config):
+    """Common setup for Vera devices."""
+    global VERA_CONTROLLER
+    import pyvera as veraApi
+
+    config = base_config.get(DOMAIN)
+    base_url = config.get('vera_controller_url')
+    if not base_url:
+        _LOGGER.error(
+            "The required parameter 'vera_controller_url'"
+            " was not found in config"
+        )
+        return False
+
+    VERA_CONTROLLER, created = veraApi.init_controller(base_url)
+
+    if created:
```

Comment:
What does it mean if not created? Should we still continue to execute the rest of this method or can we return?